src/rsz/src/Resizer.cc

```diff
@@ -624,6 +624,172 @@ void Resizer::balanceRowUsage()
 
 ////////////////////////////////////////////////////////////////
 
+// Inspect the timing arcs on a buffer size to find the capacitance
+// points on the piecewise linear delay model
+static void populateBufferCapTestPoints(LibertyCell* cell,
+                                        LibertyPort* in,
+                                        LibertyPort* out,
+                                        std::vector<float>& points)
+{
+  for (TimingArcSet* arc_set : cell->timingArcSets()) {
+    TimingRole* role = arc_set->role();
+    if (role == TimingRole::combinational() && arc_set->from() == in
+        && arc_set->to() == out) {
+      for (TimingArc* arc : arc_set->arcs()) {
+        auto model = dynamic_cast<sta::GateTableModel*>(arc->model());
+        if (model) {
+          auto dm = model->delayModel();
+          for (const sta::TableAxis* axis :
+               {dm->axis1(), dm->axis2(), dm->axis3()}) {
+            if (axis
+                && axis->variable()
+                       == sta::TableAxisVariable::
+                           total_output_net_capacitance) {
+              points.insert(
+                  points.end(), axis->values()->begin(), axis->values()->end());
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+bool Resizer::bufferSizeOutmatched(LibertyCell* worse,
+                                   LibertyCell* better,
+                                   float max_drive_resist)
+{
+  LibertyPort *win, *wout, *bin, *bout;
+  worse->bufferPorts(win, wout);
+  better->bufferPorts(bin, bout);
+
+  float extra_input_cap
+      = std::max(bin->capacitance() - win->capacitance(), 0.0f);
+  float delay_penalty = max_drive_resist * extra_input_cap;
+  float wlimit = maxLoad(network_->cell(worse));
+
+  std::vector<float> test_points;
+  populateBufferCapTestPoints(worse, win, wout, test_points);
+  populateBufferCapTestPoints(better, bin, bout, test_points);
+
+  if (test_points.empty()) {
+    return false;
+  }
+
+  std::sort(test_points.begin(), test_points.end());
+  auto last = std::unique(test_points.begin(), test_points.end());
+  test_points.erase(last, test_points.end());
+
+  for (auto p : test_points) {
+    if ((wlimit == 0 || p <= wlimit) &&
+        /* ignore high C_load/C_in region */ p
+            < std::min(bin->capacitance(), win->capacitance()) * 20.0f) {
```

Comment:
why 20?

Reply:
It's an arbitrary constant: we size cells for C_load/C_in <= 4 and buffers for C_load/C_in <= 9, and the value of 20 was chosen as something which is far enough from that. Some of these constants should probably be, at some point, replaced with a `set_opt_config` parameter

Reply:
Does it make sense to make this a constexpr variable so we can better document the purpose?

Reply:
Sure, at this point it's clear there will be other changes from review so let me do that too